src/index.tsx

```diff
@@ -115,9 +110,13 @@ export function useSpeechRecognition(options: SpeechRecognitionOptions = default
         }
       }
 
-      dispatch(setTranscript(concatTranscripts(final, interim)));
-      dispatch(setInterimTranscript(interim));
-      dispatch(setFinalTranscript(final));
+      dispatch(
+        setTranscripts({
+          transcript: concatTranscripts(final, interim),
```

Comment:
trying to decide if the concatTranscript call is a reducer concern or not.  Probably doesn't matter a whole lot one way or another but worth considering

Reply:
it's literally just concatenating things so it didn't feel like it warranted a full state transition to me as it doesn't affect the UI in any meaningful way, it just builds one of the transcripts.

Reply:
my proposal would be to change the action payload to 

```
dispatch(setTranscripts({ final, interim}))
```

then in the reducer 

```
{
 transcript: concatTranscripts(final, interim),
 final, 
 interim
}
```